lib/puppet/provider/vcsrepo/git.rb

```diff
@@ -423,16 +423,17 @@ def commits?
   # handle upstream branch changes
   # @!visibility private
   def checkout(revision = @resource.value(:revision))
+    has_changes = uncommitted_changes?
     keep_local_changes = @resource.value(:keep_local_changes)
-    stash if keep_local_changes == :true
+    stash if has_changes == :true && keep_local_changes == :true
```

Comment:
I think you want to just check truthiness like this:

```suggestion
    stash if has_changes && keep_local_changes == :true
```

`keep_local_changes` uses the `:true` symbol because it comes from the Puppet resource, I think.

Reply:
Thanks @kenyon for the quick review. I have now pushed a commit to address your suggestion.